x/metadata/client/cli/tx.go

```diff
@@ -371,21 +372,215 @@ func WriteContractSpecificationCmd() *cobra.Command {
 	return cmd
 }
 
+// WriteRecordCmd creates a command to add/update records
+func WriteRecordCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "write-record [scope-id] [record-spec-id] [name] [process] [inputs] [outputs] [parties-involved] [contract-id | session-id]",
+		Short: "Add/Update metadata record to the provenance blockchain",
+		Long: fmt.Sprintf(`Add/Update metadata record to the provenance blockchain.
+[scope-id]                 - scope metaaddress for the record
+[record-spec-id]           - associated record specification metaaddress
+[name]                     - record name
+[process]                  - comma delimited structure of process name, id (hash or bech32 address), and method: Example: processname,hashvalue,method
+[inputs]                   - semicolon delimited list of input structures.  Example: name,soure-value(hash or metaaddress),typename,status(proposed,record);...
+[outputs]                  - semicolon delimited list of outputs structures. Example: hash-value,status(pass,skip,fail);...
+[parties-involved]         - semicolon delimited list of party structures(address,role). Accepted roles: originator,servicer,investor,custodian,owner,affiliate,omnibus,provenance
+[contract-id | session-id] - a contract or session id. 
+[contract-name]            - contract name must be set if using contract id to create a new session
```

Comment:
this could probably come from the contract-specification that was referenced from `contract-id` above

Reply:
I was hoping that was the case when I first wrote it.  However, when we are doing a write session there is a validation of Session.name == contractspec.classname (https://github.com/provenance-io/provenance/blob/b29880c08e6ff856fc681ed7c83c13ca520c2483/x/metadata/keeper/session.go#L148).  I was trying to avoid the need to write a query for contract spec information in the cli command.

Reply:
> when we are doing a write session there is a validation of Session.name == contractspec.classname 
Kind of an odd situation then ... we have enough info to determine what the correct answer is for the user but we make them give us the correct answer anyway...
Is it reasonable to do a query in a cli method? I can't say I have tried that before...

Reply:
Yeah, well let me give it a try to do a query within the tx.  I wasn't sure if we wanted to do that, since I haven't seen it in our modules yet.

Reply:
> Is it reasonable to do a query in a cli method? I can't say I have tried that before...
based on looking at how query flags are handled in the CLI I would say there is a good reason why we have not seen this done before... it is not really a great idea because of the root level separation in the cli between queries and tx.